master/config.sls

```diff
@@ -36,11 +36,10 @@ include:
 {% for proxyid in salt.pillar.get('salt_master:proxy_configs:apps') %}
 /etc/systemd/system/{{ proxyid }}.service:
   file.managed:
-    - name: /etc/systemd/system/{{ proxyid }}.service
+    - name: /etc/systemd/system/{{ proxyid }}@.service
```

Comment:
For this approach we wouldn't need to loop for creating the unit template, we would write that out once and do the daemon reload. We would then loop through the defined proxies for enabling the service after the template file is on disk.